### DISCUSSION TITLE
部分字体样式的设置中，字重的显示效果在不同的系统环境下可能出现较大差异。

### DISCUSSION THREAD

**Discussion by wathper** [dyphire/mpv-config#57] [Ideas]
![image](https://github.com/dyphire/mpv-config/assets/110274576/3e4bae95-8541-4cb2-a5fe-86349778765e)

存在多个配置文件中写人了参数`Noto Sans Mono CJK SC`、`Noto Sans CJK SC`、`Noto Serif CJK SC`之后，没有指定字体的字重为`Bold`
因为在`portable_config\fonts`文件夹中仅放置有`Bold`字重的 Google 字体包
当用户 Win 系统字体中没有安装 Google 黑体、宋体时，字体样式能够正确的显示为粗体。
但是当用户安装有这两个字体的全字重时，没有指定`Bold`字重的字体样式可能会被错误的显示。

建议在醒目的地方注释进行提醒。
如果用户安装有 Google 黑体、宋体，可以将 Adobe 和 Google 合作开发的`SourceHanSansSC-Bold.otf`、`SourceHanSansHWSC-Bold.otf`（等宽）、`SourceHanSerifSC-Bold.otf`字体包放入字体文件夹。
这些字体由 Adobe 和 Google 合作开发，具备一致的显示效果。

最后用`Source Han Sans HW SC`、`Source Han Sans SC`以及`Source Han Serif SC`全局替换掉 Google 系列的对应参数。以达成和预设一致的效果。

**Comment by wathper** [comment]
不过还是建议在字体样式的设置中加入字重属性。虽然大多数人都不会在系统中安装 Google 的字体。但字幕组在编写 ASS 字幕中很可能会使用`Noto Sans CJK SC`、`Noto Serif CJK SC`字族的常规字重，如果再由字体临时挂载脚本将字幕组提供的`Noto Sans CJK SC`的常规字重读入内存、启动 MPV 播放器后，部分界面的字体样式可能也不会显示为粗体。

**Comment by dyphire** [comment]
实际上这是利用了 libass 的一个特性（bug）有意为之的，在 Windows 上当一个字体的常规字重和粗体字重使用同一 Family Name 时不同于 xysubfilter，libass 会优先加载附加的粗体字重而不是系统中的常规字重
参考 https://github.com/libass/libass/issues/509
在 libass 更改当前行为前不会针对性处理
